delegate: fix zone configs for SHOW CREATE TABLE with no partitions

otan opened the pull request:
Fixed the regression in c44619c38ba6f22d8d6590a34600ba0c2fd7fdd9 which
had no tests to cover this case.

Release note (bug fix): Fixed a bug where if the zone configurations
would not display on a SHOW CREATE TABLE if there are no partitions
even though there are zone configurations on the index or tables
themselves.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/65167)
<!-- Reviewable:end -->

jordanlewis reviewed (approved):
Thank you! 🙇

otan committed:
delegate: fix zone configs for SHOW CREATE TABLE with no partitions

Fixed the regression in c44619c38ba6f22d8d6590a34600ba0c2fd7fdd9 which
had no tests to cover this case.

Release note (bug fix): Fixed a bug where if the zone configurations
would not display on a SHOW CREATE TABLE if there are no partitions
even though there are zone configurations on the index or tables
themselves.

otan commented:
tftr!

bors r=jordanlewis
